src/main/java/parser/CarsParser.java

```diff
@@ -1,24 +1,27 @@
 package parser;
 
-import static parser.AttributeValueParser.*;
-
 import entity.Car;
 import entity.Company;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+import static parser.AttributeValueParser.COLOR;
+import static parser.AttributeValueParser.ID;
+
 public class CarsParser {
     private static final Logger LOGGER = LogManager.getLogger(Company.class);
     private static final AttributeValueParser parser = new AttributeValueParser();
     private static final String CAR_REGEX = "\\d+";
+    private static final int COMPANY_PARK_CAPACITY = 10;
 
-    public List<Car> getCars(final List<String> carList) {
-        final List<Car> cars = new ArrayList<>();
+    public BlockingQueue<Car> parseCars(final List<String> carList) {
+        final BlockingQueue<Car> cars = new ArrayBlockingQueue<Car>(COMPANY_PARK_CAPACITY);
```

Comment:
For the future I would recommend to not set the queue size if you aren't sure that parsing cars count wouldn't be the same.